core/kernel/transfer_list.c

```diff
@@ -389,13 +394,16 @@ bool transfer_list_set_data_size(struct transfer_list_header *tl,
 		    tl->size + mov_dis > tl->max_size) {
 			return false;
 		}
-		ru_new_ev = old_ev + mov_dis;
-		memmove((void *)ru_new_ev, (void *)old_ev, tl_old_ev - old_ev);
+		r_new_ev = old_ev + mov_dis;
 		tl->size += mov_dis;
-		gap = ru_new_ev - new_ev;
 	} else {
-		gap = old_ev - new_ev;
+		mov_dis = ROUNDDOWN(old_ev - new_ev,
+				    TL_ALIGNMENT_FROM_ORDER(tl->alignment));
+		r_new_ev = old_ev - mov_dis;
+		tl->size -= mov_dis;
 	}
+	gap = r_new_ev - new_ev;
+	memmove((void *)r_new_ev, (void *)old_ev, tl_old_ev - old_ev);
```

Comment:
Please add a comment explaining that we're moving eventual entries following this entry to either make room or to close the gap.

Reply:
Done.